pkg/api/types.go

```diff
@@ -2766,6 +2767,22 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// DeletePropagationPolicy decides whether and how garbage collection will be performed.
+type DeletePropagationPolicy string
+
+const (
+	// The default depends on the existing finalizers on the object and the type of the object.
+	DeletePropagationDefault DeletePropagationPolicy = "DeletePropagationDefault"
```

Comment:
Made the variable names succinct.

nil is treated as Default. Both keeping and removing DeletePropagationDefault is ok with me. @lavalamp do you prefer removing it?